### PR TITLE
Move 'defer' call before creation on various e2e tests.

### PR DESCRIPTION
**Pull request by fabioy**

### CODE REVIEW

**Comment by rjnagal** [comment]
@fabioy Is this change required to cleanup partial creations?

File: test/e2e/events.go [hunk 1]
```diff
@@ -71,15 +71,15 @@ var _ = Describe("Events", func() {
 		}
 
 		By("submitting the pod to kubernetes")
-		_, err := podClient.Create(pod)
-		if err != nil {
-			Fail(fmt.Sprintf("Failed to create pod: %v", err))
-		}
 		defer func() {
 			By("deleting the pod")
 			defer GinkgoRecover()
```
Comment (roberthbailey) [inline test/e2e/events.go]: Not really part of your CL, but GinkgoRecover is only necessary in go routines, not defer statements. Please remove this line.

File: test/e2e/events.go [hunk 2]
```diff
@@ -71,15 +71,15 @@ var _ = Describe("Events", func() {
...
 			podClient.Delete(pod.Name)
 		}()
+		_, err := podClient.Create(pod)
```
Comment (roberthbailey) [inline test/e2e/events.go]: if _, err := podClient.Create(pod); err != nil {
  Fail(...)
}

File: test/e2e/events.go [hunk 3]
```diff
@@ -71,15 +71,15 @@ var _ = Describe("Events", func() {
...
+		if err != nil {
+			Fail(fmt.Sprintf("Failed to create pod: %v", err))
```
Comment (roberthbailey) [inline test/e2e/events.go]: You can also be able to use FailF here.

File: test/e2e/secrets.go [hunk 4]
```diff
@@ -57,18 +57,16 @@ var _ = Describe("Secrets", func() {
...
 		secret, err := c.Secrets(ns).Create(secret)
-		By(fmt.Sprintf("Creating secret with name %s", secret.Name))
-		if err != nil {
-			Fail(fmt.Sprintf("unable to create test secret %s: %v", secret.Name, err))
-		}
-
-		// Clean up secret
 		defer func() {
 			By("Cleaning up the secret")
 			if err = c.Secrets(ns).Delete(secret.Name); err != nil {
```
Comment (roberthbailey) [inline test/e2e/secrets.go]: if err := ...

File: test/e2e/service.go [hunk 5]
```diff
@@ -107,15 +107,15 @@ var _ = Describe("Services", func() {
...
-		_, err := podClient.Create(pod)
-		if err != nil {
-			Fail(fmt.Sprintf("Failed to create %s pod: %v", pod.Name, err))
-		}
 		defer func() {
 			By("deleting the pod")
 			defer GinkgoRecover()
 			podClient.Delete(pod.Name)
 		}()
+		_, err := podClient.Create(pod)
```
Comment (roberthbailey) [inline test/e2e/service.go]: if _, err := ...; err != nil { }

File: test/e2e/service.go [hunk 6]
```diff
@@ -210,9 +210,7 @@ var _ = Describe("Services", func() {
 
 		validateEndpointsOrFail(c, ns, serviceName, expectedPort, []string{})
 
-		name1 := "test1"
-		addEndpointPodOrFail(c, ns, name1, labels)
-		names := []string{name1}
+		names := []string{}
```
Comment (roberthbailey) [inline test/e2e/service.go]: var names []string

**Comment by fabioy** [comment]
I've addressed all comments and fixed up a few more "Failf" changes as well. PTAL.

**Comment by fabioy** [comment]
Addressed a few more comments. PTAL.

**Comment by roberthbailey** [comment]
LGTM.

**Pull request merged**